### PR TITLE
Remove compute button and sync checkbox from the feature editor

### PR DESCRIPTION
**Pull request by lppedd**
Two points:

1. `IFeature` has a `computeImports` method, which I didn't touch because that - even if unused in this project - would be a breaking API change.
2. The help content doesn't explain what the _Compute_ button or _Recompute when..._ checkbox do, but the image example is outdated.

Closes: eclipse-pde#37

### CODE REVIEW

**Comment by lppedd** [comment]
@HannesWell FYA

**Comment by lppedd** [comment]
Not sure about the failing tests, they seem unrelated to the scope of this change, which is fair since I've also removed UI elements.

**Comment by HannesWell** [comment]
Thanks @lppedd, this looks good on the first sight, I'll review it in detail in the next days.

> Not sure about the failing tests, they seem unrelated to the scope of this change, which is fair since I've only removed UI elements.

That's probably right. We recently added workflows on more platforms and still have to fix some tests, so you don't have to worry about that.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/feature/RequiresSection.java [hunk 1]
```diff
@@ -89,7 +84,7 @@ public Object[] getElements(Object parent) {
 	}
 
 	public RequiresSection(FeatureDependenciesPage page, Composite parent) {
-		super(page, parent, Section.DESCRIPTION, new String[] {PDEUIMessages.FeatureEditor_RequiresSection_plugin, PDEUIMessages.FeatureEditor_RequiresSection_feature, PDEUIMessages.FeatureEditor_RequiresSection_remove, PDEUIMessages.FeatureEditor_RequiresSection_compute});
+		super(page, parent, Section.DESCRIPTION, new String[] {PDEUIMessages.FeatureEditor_RequiresSection_plugin, PDEUIMessages.FeatureEditor_RequiresSection_feature, PDEUIMessages.FeatureEditor_RequiresSection_remove});
 		getSection().setText(PDEUIMessages.FeatureEditor_RequiresSection_title);
 		getSection().setDescription(PDEUIMessages.FeatureEditor_RequiresSection_desc);
```
Comment (HannesWell) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/feature/RequiresSection.java:89]: I think the value of the property `FeatureEditor_RequiresSection_desc` should be rephrased to something like:
```
List Plug-ins and Features that will need to be present before installing this feature.
```

**Review by HannesWell — CHANGES_REQUESTED** [review]
> Two points:
> 
>     1. `IFeature` has a `computeImports` method, which I didn't touch because that - even if unused in this project - would be a breaking API change.

`IFeature` is not a public API, it is just an internal Interface so you can just delete the `computeImports` method and its implementation in the `Feature` class. You can then also delete all methods that are only called by the implementation. But you will probably find that out quickly but simply deleting the implementation. Most called methods are private so you should get an error/warning if they are not called anymore.

> 
>     2. The help content doesn't explain what the _Compute_ button or _Recompute when..._ checkbox do, but the image example is outdated.

Unfortunatly I cannot find the help for that part of the Editor quickly (🙄), but the doc for PDE lives in https://github.com/eclipse-platform/eclipse.platform.common/tree/master/bundles/org.eclipse.pde.doc.user.

If you want to update that as well, it would be great. :)

**Comment by HannesWell** [comment]
> @HannesWell @lppedd would be great to finish this today so it can go into M1

The M1 freeze period already started yesterday at 12pm.
@lppedd nevertheless it would great to have this for M2. :)

**Comment by lppedd** [comment]
@HannesWell @laeubi I'm busy as of now but this will be done before the end of the week.

**Comment by vogella** [comment]
> @HannesWell @laeubi I'm busy as of now but this will be done before the end of the week.

Thanks, end of week is approaching, would be nice to have this in.

**Review by HannesWell — APPROVED** [review]
@lppedd I have applied the suggested changes and pushed them to your branch.

Since this overall not a big change I plan to merge this as soon as the build succeeds so that we can have it just in time for the 2023-06 release.

**Comment by HannesWell** [comment]
@lppedd thanks for this PR.

**Pull request merged**